### DISCUSSION TITLE
Feature Request: Decommission mode - to allow transfer of ownership

### DISCUSSION THREAD

**Discussion by K1WIZ** [arendst/Tasmota#14285] [Ideas]
Hi,

I've installed a lot of Tasmotized switches in my home.  We're moving and selling the house and I'm looking for a clean way to remove all the tasmota stuff.

Since we're taking our wifi gear with us, the in-wall tasmotized switches will no longer be able to connect to the network and rather than replace a bunch of switches, we'd like to put the tasmotized switches in a "dumb mode" that allows them to operate as normal manual switches.  The problem is that when removed from the network, they tend to cause the lights to flicker off momentarily and randomly (seemingly maybe caused by repeated rebooting?).

I can't be the first person to sell a house and have tasmotized hardware.   Can someone point me in the right direction on how to properly decommission tasmotized stuff?   ie. just make it operate like a normal switch?

----

**Feature Request: Decommission mode - to allow transfer of ownership**

I recently learned when preparing my house for sale that removing the wifi network makes tasmotized switches act erratically.  ie. though I could operate a switch manually, lights on tasmotized switches would randomly flick off for a second at random intervals, perhaps caused by loss of the wifi network (which we are taking with us on our move).  

To help folks easily detach an expensive network infrastructure from a Tasmotized switch deployment, it would be useful to have the following feature and work-flow available so as not to have to rip and replace dozens of switches:

1) user could activate a "decommission" mode on all deployed switches, either by MQTT command or console command. 
2) such "decommission" mode would kill the need to constantly try a wifi connection, and would put the wifi function in disabled state, and allow the switch to function as a manual switch only.  This would allow a new owner to assume simple manual control (if desired).
3) If a new owner decided they wanted to re-activate the wifi and attach the switch to a new wifi network, they could perform a special toggle sequence to reactivate the wifi function (like the wifi manager page) and use that to attach to the new owner's network, and subsequently, home hub, etc.

Such a decommission function would be incredibly useful and allow tasmotized deployments to be easily transferred with the sale of a home, whether the new owner elects to re-attach to a new network or not.

**Comment by barbudor** [comment]
Why not removing the switches and set the original mechanical ones in place ?

Tasmota needs Wifi and may miss-behave without it so it's not a good idea to let them in place.

**Reply by K1WIZ** [reply]
There a lot of in-wall switches I'd have to replace.  I'd prefer not to do that if it can be avoided.   It would be a huge hassle and expense to not only replace all these with dumb switches (I tossed the originals, and original switchplates because all the smart switches are deco style and not traditional like the original - so I'd have to also replace all the switchplates too).

Perhaps there should be a "decommission mode" as a feature request so people can easily sell a house and revert smart switches to manual mode - perhaps with a toggle sequence that can re-enable AP mode should a new owner wish to make them smart again?   

psst...  that'd be real smart!  ;)  and less junk in landfills.  Feature request perhaps?

**Reply by jziolkowski** [reply]
Real smart thing to do is to remove the smart switches. I'm pretty sure they cost a fraction of the smart switches, so you will be losing less money as compared to leaving them behind.

**Reply by sfromis** [reply]
You could also tell the buyer: This is a smart home which you need to learn how to use, or get an electrician to change it to suit your preferences.

**Reply by K1WIZ** [reply]
> You could also tell the buyer: This is a smart home which you need to learn how to use, or get an electrician to change it to suit your preferences.

That would be fine if I'm leaving behind the network and server equipment I use to operate the tasmotized switches.  I'm not doing that as I built everything on docker containers and KVMs.  It'd just be nice to have a "decommission" mode whereby once set, makes the switch operate as a manual switch and could then be "re-animated" by a special toggle sequence.   Seems that should be possible in theory and would help people so they wouldn't have to rip and replace switches all over again.   This effectively is a logic problem, which could be solved by addition of code.   Though it may not help me in this instance, I'll open a feature request for it - in the hopes it helps others.

**Reply by jziolkowski** [reply]
> I'll open a feature request for it 

IMHO don't bother

**Reply by sfromis** [reply]
"Learning to use" includes the buyer doing networking and setup. You could leave the Tasmota devices in initial setup mode, ready to configure. While this would be a tall order for a non-techie buyer, it is their responsibility to adapt to their needs, including getting help (like from an electrician) if they are not up to the task.

**Reply by K1WIZ** [reply]
> "Learning to use" includes the buyer doing networking and setup. You could leave the Tasmota devices in initial setup mode, ready to configure. While this would be a tall order for a non-techie buyer, it is their responsibility to adapt to their needs, including getting help (like from an electrician) if they are not up to the task.

Not all potential buyers are "tech savvy" and would not want to both with all that, nor having to rip/replace, etc.   In my position as a seller of the home, (and the time and money I spent to deploy Tasmotized in-wall switches) it sucks for me, and makes me less thrilled at the prospect of doing a rip and replace should a buyer not want to by my house because they shouldn't have to bother.    Just saying, some kind of decommission function would allow much easier transfer of ownership - with no hassle to a seller or buyer.   It would just be one more awesome feature.

**Reply by sfromis** [reply]
Well, it is not much different from paint on the walls. The buyer may hate the colors, and feel a strong need to have the walls painted before wanting to live in the house. I already clearly laid out that the buyer may have to get help to adapt to their needs, which obviously includes buyers not much into tech. As a buyer of a house, I'd fully expect to have to do things to get the house to be livable by my preferences. Just a task to deal with, not a major factor in me choosing to buy that particular house.

**Reply by K1WIZ** [reply]
> Well, it is not much different from paint on the walls. The buyer may hate the colors, and feel a strong need to have the walls painted before wanting to live in the house. I already clearly laid out that the buyer may have to get help to adapt to their needs, which obviously includes buyers not much into tech. As a buyer of a house, I'd fully expect to have to do things to get the house to be livable by my preferences. Just a task to deal with, not a major factor in me choosing to buy that particular house.

not sure where you live, but people in my market would be super persnickety about something like this.   (anything electrical) painting a desired color would be within the bounds of reasonable on the other hand.  Just sayin'.

**Comment by joba-1** [comment]
Why not use a cheap raspberry pi zero w to set up a wlan AP, ntp server and mqtt broker where all devices can connect to? Adding some basic HA software to it may easily increase the value of your house beyond the investment.

I never understood why the tasmota community reacts a bit allergic to feature requests towards an isolated tasmota (like no time, no mqtt, no wlan). Just saying "there is no such feature yet, a PR would be nice though" would suffice imo?

**Reply by K1WIZ** [reply]
> Why not use a cheap raspberry pi zero w to set up a wlan AP, ntp server and mqtt broker where all devices can connect to? Adding some basic HA software to it may easily increase the value of your house beyond the investment.
> 
> I never understood why the tasmota community reacts a bit allergic to feature requests towards an isolated tasmota (like no time, no mqtt, no wlan). Just saying "there is no such feature yet, a PR would be nice though" would suffice imo?

I thought about something like that, but there are a few problems going that route:
1) the pi zero W AP would be weak, and not reliably cover the entire deployment.
2) unless it's turn-key COTS hardware/software, who would provide support in the event the new owner had an issue and was non-technical?   I wouldn't want to leave them in that situation - it would suck for them.
3) in the case of a partially tech savvy buyer, in support/etc, I want to be free and clear of the house and not have to worry about a home baked control system coming back to haunt me. 

a decommission function would make this all so graceful and simple for ownership transfer.

**Reply by ascillato** [reply]
Hi,

Let me tell you my experience on this. I have installed tasmotas in few houses and in a couple, this situation showed up. In one, the new buyer didn't want to have anything like that inside their new home. So, before the sale was made, the previous owner asked me to remove everything and leave that home as standard as possible. No cameras, no alarm, no smart devices. So for that type of buyers, leaving stale electronics is considered as unsafe and will reduce your chances for a sale a lot.
In the other home, they made the sale with all the smart devices, and with everything working. They made a small documentation on how to use the smart home and they increase the value a little from the standard price.

My recommendation to you is to not be in that situation were there is stale electronics installed consuming energy and doing nothing. I believe is the worst situation to leave a house incomplete. You should sale the house without any device, or with everything working.

**Reply by sfromis** [reply]
Indeed, devices left in walls should be understood and appreciated by the new owner. In addition to consuming energy, stale electronics does have a risk of breaking down or loosing the "decommission" config. While not a big risk, electronic devices in walls can also be bit of a fire hazard. If a buyer does not want "smart" stuff, they have a reasonable expectation of getting the simple reliability of old-fashioned "dumb" switches. The one installing "smart" stuff choose to lose this simplicity to get the extra control features. It should be either-or, there does not exist a fully viable middle ground.

**Comment by Jason2866** [comment]
To answer your feature request. Will not be done. The reasons for are explained very detailed from @ascillato
Thank you for sharing your idea.
